riptide-httpclient/src/test/java/org/zalando/riptide/httpclient/AbstractApacheClientHttpRequestFactoryTest.java

```diff
@@ -179,6 +186,27 @@ void shouldReleaseConnection() {
         });
     }
 
+    @Test
+    void shouldReleaseConnectionOnFailureToReadBody() {
+        driver.addExpectation(onRequestTo("/wrong-content-type"), giveResponse("[]", "text/plain"));
+        driver.addExpectation(onRequestTo("/wrong-content-type"), giveResponse("[]", "text/plain"));
+
+        final Runnable request = throwingRunnable(() -> {
```

Comment:
```suggestion
        final ThrowingRunnable<Exception> request = () -> {
```

Reply:
On the one hand "why not" but on the other hand "why"?

Reply:
I usually prefer to use the type over the functions. But it's actually shorter...